athena-snowflake/src/main/java/com/amazonaws/athena/connectors/snowflake/SnowflakeRecordHandler.java

```diff
@@ -86,18 +109,106 @@ public PreparedStatement buildSplitSql(Connection jdbcConnection, String catalog
         PreparedStatement preparedStatement;
         try {
             if (constraints.isQueryPassThrough()) {
-            preparedStatement = buildQueryPassthroughSql(jdbcConnection, constraints);
-        }
-        else {
-            preparedStatement = jdbcSplitQueryBuilder.buildSql(jdbcConnection, null, tableNameInput.getSchemaName(), tableNameInput.getTableName(), schema, constraints, split);
-        }
+                preparedStatement = buildQueryPassthroughSql(jdbcConnection, constraints);
+            }
+            else {
+                preparedStatement = jdbcSplitQueryBuilder.buildSql(jdbcConnection, null, tableNameInput.getSchemaName(), tableNameInput.getTableName(), schema, constraints, split);
+            }
 
-        // Disable fetching all rows.
-        preparedStatement.setFetchSize(FETCH_SIZE);
+            // Disable fetching all rows.
+            preparedStatement.setFetchSize(FETCH_SIZE);
         }
         catch (SQLException e) {
             throw new AthenaConnectorException(e.getMessage(), ErrorDetails.builder().errorCode(FederationSourceErrorCode.INTERNAL_SERVICE_EXCEPTION.toString()).build());
         }
         return preparedStatement;
     }
+
+    @Override
+    protected CredentialsProvider getCredentialProvider()
```

Comment:
This looks duplicate. Avoid duplicate code logic, instead define it in the utility class and call it.